src/Analyzers/CSharp/CodeFixes/UseInterpolatedString/CSharpUseInterpolatedStringCodeFixProvider.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.UseInterpolatedString;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseInterpolatedString;
+
+[ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.UseInterpolatedString), Shared]
+[method: ImportingConstructor]
+[method: SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+internal partial class CSharpUseInterpolatedStringCodeFixProvider() : AbstractUseInterpolatedStringCodeFixProvider
+{
+    protected override async Task FixOneAsync(Document document, Diagnostic diagnostic, SyntaxEditor editor, CancellationToken cancellationToken)
+    {
+        var span = diagnostic.Location.SourceSpan;
+        var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+        var node = root.FindNode(span, getInnermostNodeForTie: true) as LiteralExpressionSyntax;
+        if (node?.Span != span || !node.IsKind(SyntaxKind.StringLiteralExpression))
+            return;
+
+        var newNode = CreateInterpolatedString(node);
+        editor.ReplaceNode(node, newNode);
+    }
+
+    private static InterpolatedStringExpressionSyntax CreateInterpolatedString(SyntaxNode stringLiteralExpressionNode)
```

Comment:
Given my suggestion above you should be able to strongly type parameter node here as well:
```suggestion
    private static InterpolatedStringExpressionSyntax CreateInterpolatedString(StringLiteralExpressionSyntax stringLiteralExpressionNode)
```

Reply:
I see, you've already changed this parameter type to `LiteralExpressionSyntax`, good. Feedback resolved